lib/gitlab/ci/templates/pipeline/NpmJestCoveragePipeline.yml

```diff
@@ -0,0 +1,57 @@
+stages:
+  - build
+  - test
+  - deploy
+
+build:
+  image: node:16
+  stage: build
+  script:
+    - npm install
+    - npm run build
+  cache:
+    paths:
+      - node_modules/
+  artifacts:
+    expire_in: 1 days
+    when: on_success
+    paths:
+      - node_modules/
+
+
+# Run Tests
+test:
+  image: node:16
+  stage: test
+  coverage: '/Lines\s*:\s*(\d+.?\d*)%/'
+  dependencies:
+    - build
+  script:
+    - npm run test:ci
+  cache:
+    paths:
+      - coverage/
+  artifacts:
+    paths:
+      - coverage/
+    when: always
+    reports:
+      junit:
+        - junit.xml
+      coverage_report:
+        coverage_format: cobertura
+        path: coverage/cobertura-coverage.xml
+
+pages:
+  stage: deploy
+  dependencies:
+    - test
+  script:
+    - mkdir .public
+    - cp -r coverage/* .public
+    - mv .public/* public/
+  artifacts:
+    paths:
+      - public
+  only:
+    - main
```

Comment:
We don't use main so this will be an issue. We also generally want to make this configurable through a variable so projects can configure what they need. Also, `only` is old syntax and is not recommended anymore; use `rules` instead

Reply:
I've switched this out with the below based on what I found in the java gradle template. 

```
rules:
    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
```

Are those two variables things that we get for free in gitlab or does the repo still need to define them?

Reply:
nm, I found them in the gitlab docs.